com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/certficate_checker/DefaultMtaCertificateChecker.java

```diff
@@ -0,0 +1,56 @@
+package com.sap.cloud.lm.sl.cf.process.util.certficate_checker;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Collection;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.sap.cloud.lm.sl.cf.process.Constants;
+import com.sap.cloud.lm.sl.cf.process.util.verifier.Verifier;
+import com.sap.cloud.lm.sl.common.SLException;
+
+@Component
+public class DefaultMtaCertificateChecker implements MtaCertificateChecker {
+
+    @Inject
+    private Verifier verifier;
+
+    @Override
+    public void checkCertificates(URL url, List<X509Certificate> certificates, String certificateCN) {
+        verifier.verify(certificates, certificateCN, url);
+    }
+
+    @Override
+    public List<X509Certificate> readProvidedCertificates(String certificatesFilename) {
+        InputStream symantecInputStream = getClass().getResourceAsStream(certificatesFilename);
```

Comment:
Can we put this in a try-with-resources block?

Reply:
Yes, we can. Done